apps/tools/scxmldump/main.cpp

```diff
@@ -50,6 +50,17 @@ static void removeAllArrivals(Seiscomp::DataModel::Origin *origin) {
 }
 
 
+static void removeUnusedArrivals(Seiscomp::DataModel::Origin *origin) {
+	for ( size_t i = 0; i < origin->arrivalCount(); ) {
+		auto arrival = origin->arrival(i);
+		if (arrival->weight() == 0) {
```

Comment:
The weight of an arrival is returned by the locator after considering all arrivals for locating. How this weight is generated depends on the locator itself and it depends on all provided arrivals including the arrival which is returned with a weight of 0. Therefore, removing the arrival with weight = 0 may not be the right choice according to what the option suggests: "Do not export unused origin arrivals.".

Instead of comparing the weight, a more appropriate may be  to filter out all arrivals where neither time, slowness nor backazimuth are used. But even then, ignoring these arrivals may cause later distortions since some SeisComP modules such as scevent can make use of such arrivals. Therefore, I recommend to refrain from this option and to accept all arrivals or origins.